snarkos/main.rs

```diff
@@ -102,39 +108,31 @@ async fn start_server(config: Config) -> anyhow::Result<()> {
         Duration::from_secs(config.p2p.peer_sync_interval.into()),
     )?;
 
+    info!("Loading storage at '{}'...", path.to_str().unwrap_or_default());
+    let storage = Arc::new(KeyValueStore::new(RocksDb::open(&path)?));
+    info!("Storage finished loading");
+
     // Construct the node instance. Note this does not start the network services.
     // This is done early on, so that the local address can be discovered
     // before any other object (miner, RPC) needs to use it.
-    let mut node = Node::new(node_config)?;
-
-    let is_storage_in_memory = LedgerStorage::IN_MEMORY;
-
-    let storage = if is_storage_in_memory {
-        Arc::new(MerkleTreeLedger::<LedgerStorage>::new_empty(
-            None::<std::path::PathBuf>,
-        )?)
-    } else {
-        info!("Loading storage at '{}'...", path.to_str().unwrap_or_default());
-        Arc::new(MerkleTreeLedger::<LedgerStorage>::open_at_path(path.clone())?)
-    };
-    info!("Storage finished loading");
+    let mut node = Node::new(node_config, storage.clone()).await?;
 
     // For extra safety, validate storage too if a trim is requested.
-    if config.storage.validate || config.storage.trim {
-        let now = std::time::Instant::now();
-        storage
-            .validate(None, snarkos_storage::validator::FixMode::Everything)
-            .await;
-        info!("Storage validated in {}ms", now.elapsed().as_millis());
-        if !config.storage.trim {
-            return Ok(());
-        }
-    }
+    // if config.storage.validate || config.storage.trim {
+    //     let now = std::time::Instant::now();
+    //     storage
+    //         .validate(None, snarkos_storage::validator::FixMode::Everything)
+    //         .await;
+    //     info!("Storage validated in {}ms", now.elapsed().as_millis());
+    //     if !config.storage.trim {
+    //         return Ok(());
+    //     }
+    // }
```

Comment:
note for other reviewers: once this PR is ready, I will update the storage validator; until then, an older binary can be used to validate the node, as the on-disk format doesn't change yet